### PR TITLE
Prepend sass variables with "hds-" (HDS-2974)

### PR DESCRIPTION
**Pull request by KristinLBradley**
### :pushpin: Summary
If merged, this PR will prepend Sass variables used in styles with "hds-".

### :link: External links

- Jira ticket: [HDS-2974](https://hashicorp.atlassian.net/browse/HDS-2974)

***

### 👀 Component checklist

- [x] Percy was checked for any visual regression
- [x] A changelog entry was added via [Changesets](https://github.com/changesets/changesets) if needed (see [templates here](https://github.com/hashicorp/design-system/blob/main/wiki/Website-Changelog.md#templates-for-npm-packages))

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-2974]: https://hashicorp.atlassian.net/browse/HDS-2974?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by KristinLBradley** [comment]
Let me know if I either missed any or went overboard anywhere.

**Review by didoo — COMMENTED** [review]
@KristinLBradley great job

I noticed that there also some variables that are used as maps, that could be renamed too (you can easily find them if you search the folder `./design-system/packages/components/app/styles` using the regex `^\$((?!hds-)[a-zA-Z_-]+)`)
screenshot_3421

probably I would rename them something like this:
`$size-props` in `badge-count.scss` → `$hds-badge-count-size-props` (verbose, but semantic and unique)

what do you think?

**Comment by KristinLBradley** [comment]
@didoo Would this be considered a "minor" change as consumers aren't using these variables directly? (or they shouldn't be)

https://hashicorp.atlassian.net/browse/HDS-2874

**Review by didoo — APPROVED** [review]
Seems good to me 👍 (I haven't checked the rendered showcase pages, but since the Percy tests are passing I assume everything is working as expected)

**Comment by KristinLBradley** [comment]
> Seems good to me 👍 (I haven't checked the rendered showcase pages, but since the Percy tests are passing I assume everything is working as expected)

Thanks! Percy showed no changes in the rendered pages.

Would this be considered a "minor" change as [consumers aren't using these variables directly](https://hashicorp.atlassian.net/browse/HDS-2874)? I need to add a changelog still.

**Comment by didoo** [comment]
> Would this be considered a "minor" change as [consumers aren't using these variables directly](https://hashicorp.atlassian.net/browse/HDS-2874)? I need to add a changelog still.

Since these are internal variables, probably a patch is OK

File: .changeset/gold-eagles-complain.md
Comment (KristinLBradley) [inline .changeset/gold-eagles-complain.md:1]: Is this description for the changes ok or should I add any other details?

**Review by alex-ju — APPROVED** [review]

File: .changeset/gold-eagles-complain.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@hashicorp/design-system-components": patch
+---
+
+Added "hds-" prefix to Sass variables for component styles where missing.
```
Comment (didoo) [inline .changeset/gold-eagles-complain.md:5]:
```suggestion
Added `hds-` prefix to Sass variables for component styles (where missing).
```

**Review by didoo — APPROVED** [review]

**Pull request merged**